src/documents/jqUnit.md

```diff
@@ -0,0 +1,305 @@
+    <div id="qunit-testrunner-toolbar"></div>
+    <h2 id="qunit-userAgent"></h2>
+    <ol id="qunit-tests"></ol>
+
+    <!-- Test HTML -->
+    <div id="qunit-fixture">
+    </div>
+```
+
+Mysteriously, these UI element ids and their functions are not documented on the QUnit site itself - although you can find explanations of them in some 3rd party tutorials.
+Any markup used within test fixtures must be placed within the div with id `qunit-fixture` - QUnit will take care of tearing this down and restoring it in its
+original condition before the start of every test.
+
+Your test fixtures should be scheduled to start only once this markup has loaded - you can achieve this by either starting them within a [$(document).ready](https://api.jquery.com/ready/)
+callback, or by writing a `<script>` block at the base of your HTML file which starts them.
+
+`jqUnit` provides a special set of assertions and utilities for testing within the browser environment:
+
+### jqUnit.assertNode (message, expected, node)
+
+Checks a subtree of DOM nodes descended for a particular node against a condensed JSON representation, allowing multiple aspects of a rendered UI to be checked in a single
+operation. Each attribute of each DOM node is attached as a direct property, the node's tag name is attached as `nodeName`, the node's element test is attached as `nodeText`, and then
+all children are recursively attached as the property `children`. In addition, at any stage, the entire nested markup may be attached as `nodeHTML`.
+
+As an example, the following markup:
+
+```html
+    <a href="a-link"><img src="a-source"/></a>
+```
+
+will compare equal to
+
+```json
+   {
+       nodeName: "a",
+       href: "a-link",
+       children: [
+           {
+           nodeName: "img",
+           src: "a-source"
+           }
+       ]
+   }
+```
+* `expected {Object|Array}` A condensed JSON respresentation of a set of assertions to make about a subtree of DOM nodes, or an array of these
+* `node {DOM|Array of DOM|jQuery}` The DOM node to be checked against `expected` - this may also be an array of DOM nodes or a jQuery object.
+
+
+### jqUnit.assertDomEquals(message, expected, actual)
+
+Compare two lists of DOM elements (or jQueries) for being equal by virtue of containing the same DOM elements in the same order. This will
+be achieved by canonicalising the DOM elements onto their ids by means of the canonicalisation function `jqUnit.canonicaliseDom`
+
+* `expected {Array of DOM|jQuery}` The expected list of DOM nodes
+* `actual {Array of DOM|jQuery}` The actual list of DOM nodes to be compared with `expected`.
+
+### jqUnit.isVisible(msg, selector)
+
+Asserts that the DOM nodes identified by `selector` are visible, in terms of matching the jQuery [`:hidden`](https://api.jquery.com/hidden-selector/) pseudoselector.
+
+* `selector: {jQueryable}` The selector or other jQueryable identifying the DOM nodes to be tested for visibility
+
+### jqUnit.notVisible(msg, selector)
```

Comment:
Are isVisible and noVisible properly describing match/not match of :hidden? or do they have it backward?